cumulusci/core/dependencies/dependencies.py

```diff
@@ -0,0 +1,910 @@
+
+                return commit.sha, ManagedPackageDependency(
+                    version_id=version_id, package_name=package_name
+                )
+
+        context.logger.warn(
+            f"No 2GP package version located for release {release_id} on {repo.clone_url}."
+        )
+        return (None, None)
+
+
+class GitHubPreviousReleaseBranch2GPResolver(GitHubReleaseBranch2GPResolver):
+    name = "GitHub Previous Release Branch 2GP Resolver"
+    branch_depth = 3
+
+
+class GitHubReleaseBranchExactMatch2GPResolver(
+    GitHubPackageDataMixin, GitHubReleaseBranchMixin, Resolver
+):
+    name = "GitHub Exact-Match 2GP Resolver"
+
+    def resolve(
+        self, dep: GitHubDynamicDependency, context: BaseProjectConfig
+    ) -> Tuple[Optional[str], Optional[ManagedPackageDependency]]:
+        release_id = self.get_release_id(context)
+
+        repo = context.get_repo_from_url(dep.github)
+        if not repo:
+            raise DependencyResolutionError(f"Unable to access repository {dep.github}")
+
+        try:
+            remote_branch_prefix = find_repo_feature_prefix(repo)
+            remote_2gp_context = find_repo_2gp_context(repo)
+        except Exception:
+            context.logger.info(
+                f"Could not find feature branch prefix or 2GP context for {repo.clone_url}. Unable to resolve 2GP packages."
+            )
+            return (None, None)
+
+        # Attempt exact match
+        try:
+            branch = get_feature_branch_name(
+                context.repo_branch, context.project__git__prefix_feature
+            )
+            release_branch = repo.branch(f"{remote_branch_prefix}{branch}")
+        except Exception:
+            context.logger.info(f"Exact-match branch not found for {repo.clone_url}.")
+            return (None, None)
+
+        version_id, commit = self.locate_2gp_package_id(
+            repo,
+            release_branch,
+            remote_2gp_context,
+        )
+
+        if version_id:
+            context.logger.info(
+                f"Located 2GP package version {version_id} for release {release_id} on {repo.clone_url} at commit {release_branch.commit.sha}"
+            )
+
+            package_name, _ = self._get_package_data(repo, commit.sha)
+
+            return commit.sha, ManagedPackageDependency(
+                version_id=version_id, package_name=package_name
+            )
+
+        context.logger.warn(
+            f"No 2GP package version located for release {release_id} on {repo.clone_url}."
+        )
+        return (None, None)
+
+
+RESOLVER_CLASSES = {
+    DependencyResolutionStrategy.STRATEGY_STATIC_TAG_REFERENCE: GitHubTagResolver,
+    DependencyResolutionStrategy.STRATEGY_2GP_EXACT_BRANCH: GitHubReleaseBranchExactMatch2GPResolver,
+    DependencyResolutionStrategy.STRATEGY_2GP_RELEASE_BRANCH: GitHubReleaseBranch2GPResolver,
+    DependencyResolutionStrategy.STRATEGY_2GP_PREVIOUS_RELEASE_BRANCH: GitHubPreviousReleaseBranch2GPResolver,
```

Comment:
I think we should probably not use 2gp in the name for these resolvers that look for a version_id in a particular commit status on a particular branch. 2gp describes the end result rather than the strategy. Once we have full support for 2gp as a project's primary release vehicle, we'll need both these strategies, and a strategy which looks for the version_id in an annotation on the latest release tag.

Reply:
Great point. I'll revise the messaging for these.

Reply:
I revised these throughout to use "Commit Status" instead of "2GP". Some internal code still refers to 2GPs.